modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/hint/GeoShapeESHintQueryBuilder.java

```diff
@@ -40,10 +40,10 @@
      */
     @Override
     public QueryBuilder make(EsHint hint, String fieldName, Object value) {
-        String[] values = validate(value, 4, "shapeId, type, index and path");
-        return QueryBuilders.geoShapeQuery(fieldName, values[0], values[1])
+        String[] values = validate(value, 3, "shapeId, index and path");
+        return QueryBuilders.geoShapeQuery(fieldName, values[0])
                             .relation(ShapeRelation.WITHIN)
-                            .indexedShapeIndex(values[2])
-                            .indexedShapePath(values[3]);
+                            .indexedShapeIndex(values[1])
+                            .indexedShapePath(values[2]);
```

Comment:
Shouldn't we keep a backward compatibility mechanism for the existing queries with 4 parameters written by users?

Reply:
yes, you are right, I will add the "type" back here saying it is unused.